### PR TITLE
Add activate existing encrypted volume 

### PR DESCRIPTION
**Pull request by richiejp**
When the variable ENCRYPTED_CANCEL_EXISTING is set, it will cancel the
activate encrypted volume prompt which appears during installation to a
storage device with existing encrypted lvm volume. A workaround is
implemented for bsc#989770 which causes the activation prompt to be
displayed twice.

When the variable ENCRYPTED_ACTIVATE_EXISTING is set it will enter the
password for the existing volume to activate it.

### CODE REVIEW

File: tests/installation/encrypted_volume_activation.pm [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use base "basetest";
```
Comment (okurz) [inline tests/installation/encrypted_volume_activation.pm]: please add copyright headers (copy from other file).

File: tests/installation/encrypted_volume_activation.pm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use strict;
+use testapi;
+
+sub run {
+    assert_screen 'encrypted_volume_activation_prompt';
+    if (get_var('ENCRYPT_CANCEL_EXISTING')) {
+	send_key 'alt-c', 5;
```
Comment (okurz) [inline tests/installation/encrypted_volume_activation.pm]: where did you get this from? see https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1068. In short, the `5` does not make sense

File: tests/installation/encrypted_volume_activation.pm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	if (check_screen('encrypted_volume_activation_prompt')) {
+	    record_soft_failure 'bsc#989770';
+	    send_key 'alt-c';
+	}
```
Comment (okurz) [inline tests/installation/encrypted_volume_activation.pm]: you can shorten the wait time by calling `assert_screen` with multiple tags, i.e. in this case the next tag that should be expected afterwards in "good" case as well as the 'encrypted_volume_activation_prompt' and use `match_has_tag` to check if the prompt reappears but this will work for now.

File: tests/installation/encrypted_volume_activation.pm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    } elsif (get_var('ENCRYPT_ACTIVATE_EXISTING')) {
+	send_key 'alt-p';
+	assert_screen 'encrypted_volume_password_prompt', 10;
```
Comment (okurz) [inline tests/installation/encrypted_volume_activation.pm]: You should rely on at least the default timeout of 30 seconds, i.e. just write without explicitly stating the timeout. The assert_screen will successfully return as soon as the tag is found so no time is wasted. Slower values should only be used if you are testing a SUT behaviour itself which times out, e.g. grub boot prompt that is booting default after 8 seconds.

**Comment by okurz** [comment]
see inline comments.

Please provide a link or screenshot to a local verification run.

what about a corresponding TW test? Could you check if it works with a current TW snapshot, too, and then add to the main.pm there.

File: tests/installation/encrypted_volume_activation.pm (hunk 1 above)
Comment (sysrich) [inline tests/installation/encrypted_volume_activation.pm]: please make sure you use the version that mentions "(c)2016 SUSE Linux GmbH" ;)

File: tests/installation/encrypted_volume_activation.pm (hunk 2 above)
Comment (sysrich) [inline tests/installation/encrypted_volume_activation.pm]: And calling wait_idle seems silly when there is a check_screen immediately after, so openQA is going to take it's time before moving onto it's next actions evaluating the check_screen anyway

Comment (richiejp) [inline tests/installation/encrypted_volume_activation.pm]: I miss interpreted do_wait as a timeout value, although I did look at the source code, so I have no excuse. The reason it is there in the first place though is because I was worried about getting a false positive. I am not sure if you are saying that will definitely not happen because openQA is too slow and will not move on to check_screen fast enough?

Comment (okurz) [inline tests/installation/encrypted_volume_activation.pm]: well, after the previous assert_screen you can be sure the screen is shown. Then alt-c is pressed and the following check_screen will continue to look at the screen for 30 seconds and check if the prompt reappears. I don't see this send_key critical but the choice of timeout on the next screen can be. That is also why I suggested in the next comment to better do an assert_screen with multiple tags and not a single check screen.

File: tests/installation/encrypted_volume_activation.pm [hunk 5]
```diff
@@ -1,27 +1,38 @@
...
+    }
+    elsif (get_var('ENCRYPT_ACTIVATE_EXISTING')) {
+        send_key 'alt-p';
+        assert_screen 'encrypted_volume_password_prompt';
+        type_password;
+        send_key 'ret';
     }
-}
-
-sub test_flags {
```
Comment (richiejp) [inline tests/installation/encrypted_volume_activation.pm]: Opps, need to change base to get rid of this. Will do so in next roll up.

File: products/sle/main.pm [hunk 6]
```diff
@@ -442,6 +445,9 @@ sub load_inst_tests() {
                 loadtest "installation/user_settings.pm";
             }    # sles4sap wizard installation doesn't have user_settings step
         }
+        elsif (get_var('IMPORT_USER_DATA')) {
+            loadtest 'installation/user_import.pm';
+        }
```
Comment (okurz) [inline products/sle/main.pm]: This looks unrelated. Are you already preparing the next test?

**Comment by okurz** [comment]
* the "user import" part looks unrelated
* add a local verification run link or a screenshot showing the details overview

File: products/sle/main.pm (hunk 6 above)
Comment (richiejp) [inline products/sle/main.pm]: You suggested optionally importing user data in the progress ticket, so that is what it does. I did not find any existing test for importing user data so created a new one.

Comment (okurz) [inline products/sle/main.pm]: right, good.

**Comment by okurz** [comment]
saw the verification run locally, looks very clean :-)

Now please make sure the test_suites setting these new variables are added to osd (openqa.suse.de). In this case I will do it :-)

Can you try that on Tumbleweed, too, and add to products/opensuse/main.pm?

**Comment by richiejp** [comment]
Good, Tumbleweed is mostly done too.
